File: apps/desktop/fix-db.js
const Database = require('better-sqlite3');
const path = require('path');
const os = require('os');

const dbPath = path.join(os.homedir(), 'AppData', 'Roaming', 'pharmaflow-desktop', 'pharmaflow.db');
console.log('Opening database at:', dbPath);
const db = new Database(dbPath);

console.log('Fixing sales table records...');
const info1 = db.prepare(`UPDATE sales SET status = 'saved' WHERE status = 'completed'`).run();
console.log(`Updated ${info1.changes} sales records.`);

console.log('Fixing sync_queue payloads for Sale...');
const info2 = db.prepare(`
  UPDATE sync_queue 
  SET payload = json_replace(payload, '$.status', 'saved')
  WHERE table_name = 'Sale' AND json_extract(payload, '$.status') = 'completed'
`).run();
console.log(`Updated ${info2.changes} sync_queue payloads.`);

console.log('Resetting failed/dead-lettered sync_queue records for retry...');
const info3 = db.prepare(`
  UPDATE sync_queue
  SET is_synced = 0, retry_count = 0, next_retry_at = NULL, sync_error = NULL
  WHERE is_synced = 2 OR (is_synced = 0 AND sync_error IS NOT NULL)
`).run();
console.log(`Reset ${info3.changes} sync_queue records.`);

db.close();

File: apps/desktop/fix-db.py
import sqlite3
import os
import json

db_path = os.path.join(os.path.expanduser("~"), 'AppData', 'Roaming', 'pharmaflow-desktop', 'pharmaflow.db')
print("Opening database at:", db_path)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

print("Fixing SaleReturnItem and PurchaseReturnItem payloads...")
cursor.execute("SELECT id, payload FROM sync_queue WHERE table_name IN ('SaleReturnItem', 'PurchaseReturnItem')")
rows = cursor.fetchall()
updated = 0
for row in rows:
    row_id, payload_str = row
    try:
        payload = json.loads(payload_str)
        changed = False
        for field in ['qty', 'mrp', 'ptr', 'salePrice', 'netAmount']:
            if field in payload and isinstance(payload[field], str):
                try:
                    payload[field] = float(payload[field])
                    changed = True
                except ValueError:
                    pass
        if changed:
            new_payload_str = json.dumps(payload)
            cursor.execute("UPDATE sync_queue SET payload = ? WHERE id = ?", (new_payload_str, row_id))
            updated += 1
    except Exception as e:
        print("Error processing row", row_id, e)

print(f"Updated {updated} sync_queue payloads.")

print("Resetting failed/dead-lettered sync_queue records for retry...")
cursor.execute("UPDATE sync_queue SET is_synced = 0, retry_count = 0, next_retry_at = NULL, sync_error = NULL WHERE is_synced = 2 OR (is_synced = 0 AND sync_error IS NOT NULL)")
print(f"Reset {cursor.rowcount} sync_queue records.")

conn.commit()
conn.close()
